contracts/test/integration/vip/vip4.sol

```diff
@@ -0,0 +1,186 @@
+
+    function getMainnetProposal()
+        public
+        pure
+        override
+        returns (TimelockSimulation.action[] memory proposal)
+    {
+        proposal = new TimelockSimulation.action[](8);
+
+        proposal[0].target = MainnetAddresses.PCV_GUARD_ADMIN;
+        proposal[0].value = 0;
+        proposal[0].arguments = abi.encodeWithSignature(
+            "revokePCVGuardRole(address)",
+            MainnetAddresses.REVOKED_EOA_1
+        );
+        proposal[0]
+            .description = "Revoke PCV Guard role from revoked EOA1 by calling PCVGuardAdmin";
+
+        proposal[1].target = MainnetAddresses.CORE;
+        proposal[1].value = 0;
+        proposal[1].arguments = abi.encodeWithSignature(
+            "revokeGuardian(address)",
+            MainnetAddresses.EOA_1
+        );
+        proposal[1].description = "Revoke EOA1 as a guardian";
+
+        proposal[2].target = MainnetAddresses.CORE;
+        proposal[2].value = 0;
+        proposal[2].arguments = abi.encodeWithSignature(
+            "revokeMinter(address)",
+            MainnetAddresses.GRLM
+        );
+        proposal[2]
+            .description = "Revoke Global Rate Limited Minter's mint capability";
+
+        proposal[3].target = MainnetAddresses.CORE;
+        proposal[3].value = 0;
+        proposal[3].arguments = abi.encodeWithSignature(
+            "revokePCVController(address)",
+            MainnetAddresses.NC_PSM
+        );
+        proposal[3]
+            .description = "Revoke Non Custodial PSM PCV Controller role";
+
+        proposal[4].target = MainnetAddresses.CORE;
+        proposal[4].value = 0;
+        proposal[4].arguments = abi.encodeWithSignature(
+            "revokeGuardian(address)",
+            MainnetAddresses.GOVERNOR
+        );
+        proposal[4].description = "Revoke Guardian Role from Multisig";
+
+        proposal[5].target = MainnetAddresses.PCV_GUARD_ADMIN;
+        proposal[5].value = 0;
+        proposal[5].arguments = abi.encodeWithSignature(
+            "grantPCVGuardRole(address)",
+            MainnetAddresses.EOA_1
+        );
+        proposal[5].description = "Grant EOA 1 PCV Guard Role";
+
+        proposal[6].target = MainnetAddresses.PCV_GUARD_ADMIN;
+        proposal[6].value = 0;
+        proposal[6].arguments = abi.encodeWithSignature(
+            "grantPCVGuardRole(address)",
+            MainnetAddresses.EOA_2
+        );
+        proposal[6].description = "Grant EOA 2 PCV Guard Role";
+
+        proposal[7].target = MainnetAddresses.PCV_GUARD_ADMIN;
+        proposal[7].value = 0;
+        proposal[7].arguments = abi.encodeWithSignature(
+            "grantPCVGuardRole(address)",
+            MainnetAddresses.EOA_3
+        );
+        proposal[7].description = "Grant EOA 3 PCV Guard Role";
+    }
+
+    function mainnetSetup() public override {
+        vm.prank(MainnetAddresses.GOVERNOR);
+        Core(MainnetAddresses.CORE).grantGovernor(
+            MainnetAddresses.TIMELOCK_CONTROLLER
```

Comment:
Your timelock doesn't already have GOVERNOR?

Reply:
The new timelock does not. Hence the multisig action to grant governor to these new timelocks on both chains.